test/NuGet.Core.Tests/NuGet.Protocol.Tests/Plugins/PluginDiscovererTests.cs

```diff
@@ -738,5 +677,50 @@ public void IsExecutable_FileWithSpace_ReturnsTrue()
         }
 
 #endif
+
+        private static void SetFileExecutable(string filePath, bool executable)
+        {
+            if (!File.Exists(filePath))
+            {
+                throw new FileNotFoundException($"File not found: {filePath}");
+            }
+#if NET8_0_OR_GREATER
+            try
+            {
+                File.SetUnixFileMode(filePath, executable ?
+                    UnixFileMode.UserExecute | UnixFileMode.UserRead | UnixFileMode.UserWrite :
+                    UnixFileMode.UserRead | UnixFileMode.UserWrite);
+            }
+            catch (Exception ex)
+            {
+                throw new InvalidOperationException($"Error while setting file mode for: {filePath}", ex);
+            }
+#else
+            try
+            {
+                using (var process = new Process())
```

Comment:
Would it make sense to use CommandRunner.Run() instead?  Its pretty well tested and has retries if needed.

https://github.com/NuGet/NuGet.Client/blob/63c54d0b151e6e3c7690be3931b10f873665054f/test/TestUtilities/Microsoft.Internal.NuGet.Testing.SignedPackages/ChildProcess/CommandRunner.cs?plain=1#L30C1-L30C323